### PR TITLE
Stringless newlex

### PR DESCRIPTION
**Pull request by ZacLN**
some fixes and new lex_identifier

### CODE REVIEW

File: src/lexer.jl [hunk 1]
```diff
@@ -21,8 +24,8 @@ end
 ishex(c::Char) = isdigit(c) || ('a' <= c <= 'f') || ('A' <= c <= 'F')
 iswhitespace(c::Char) = Base.UTF8proc.isspace(c)
 
-type Lexer{IO_t <: IO, T <: AbstractToken}
-    io::IO_t
+type Lexer{T}
```
Comment (KristofferC) [inline src/lexer.jl]: What about `IOStream`?

Comment (ZacLN) [reply src/lexer.jl]: wait, I'm confused Lexers are parameterised only by Token type now right?

Comment (KristofferC) [reply src/lexer.jl]: Not really, they can be any `IO` type, so you could `tokenize(open("path"))`. for example.

Comment (ZacLN) [reply src/lexer.jl]: righto

**Comment by ZacLN** [comment]
This is junk now which I can delete I think?

**Comment by KristofferC** [comment]
I think so yes.

**Pull request closed without merging**